Sources/FireKit/FireObjectManager.swift

```diff
@@ -10,17 +10,25 @@ import FirebaseFirestore
 import FirebaseFirestoreSwift
 import SwiftUI
 
-enum PTObserverError: Error {
+enum FireError: Error {
     case decodeError
+    case indexOutOfBoundsError
+    case nilIDError
+    case nilReferenceError
+    case nilQueryError
 }
 
-public class FirestoreObjectObserver<T: Codable>: ObservableObject {
+public protocol FireIdentifiable {
```

Comment:
Unnecessary, bknz "Identifiable" in swift

Reply:
nvm optional olması lazım ama bu approach da çok iyi değil will think

Reply:
aynen denedim de optional yapmayi bi hata oldu galiba